src/components/Dialogs/BulkRenameFilesDialog.vue

```diff
@@ -0,0 +1,361 @@
+      }
+      if (foundRow.type === 'folder') {
+        folderCheckChange(foundRow as ItemRow<TreeFolder>)
+      }
+    }
+  })
+  detectIndeterminate(props.node)
+  // unfold when selected
+  if (item.selected) {
+    item.show = true
+    item.folded = false
+  }
+  dryRunRename()
+}
+
+const fileCheckChange = (item: ItemRow) => {
+  detectIndeterminate(props.node)
+  dryRunRename([item])
+}
+
+const dryRunRename = async (partialItems?: ItemRow[]) => {
+  await form.value?.validate()
+  if (!isFormValid.value) {
+    return
+  }
+  let regexp: RegExp
+  try {
+    regexp = new RegExp(regexpInput.value, regexpFlagsInput.value.join(''))
+  } catch {
+    return
+  }
+  ;(partialItems ? partialItems : items).forEach(item => {
+    if (item.type === 'file') {
+      if (item.selected && regexp.test(item.name)) {
+        item.targetName = item.name.replace(regexp, targetInput.value)
+        item.targetFullName = (item.parentItem!.fullName === '' ? '' : item.parentItem!.fullName + '/') + item.targetName
+      } else {
+        item.targetName = undefined
+        item.targetFullName = undefined
+      }
+      item.notChanged = item.name === item.targetName
+    }
+  })
+  hasDuplicated.value = false
+  const allTarget = new Map()
+  items
+    .filter(item => !!item.targetFullName)
+    .forEach(item => {
+      allTarget.set(item.targetFullName, (allTarget.get(item.targetFullName) || 0) + 1)
+    })
+  items.forEach(item => {
+    item.duplicated = allTarget.get(item.targetFullName) > 1
+    if (item.duplicated) {
+      hasDuplicated.value = true
+    }
+  })
+}
+
+const run = async () => {
+  if (!candidateItems.value.length) {
+    return toast.warn(t('dialogs.bulkRenameFiles.nothing_to_do'))
+  }
+  const reqList = []
+  for (const item of candidateItems.value) {
+    reqList.push(contentStore.renameTorrentFile(props.hash, item.fullName, item.targetFullName!))
+  }
+  running.value = true
+  Promise.all(reqList)
+    .then(() => {
+      toast.success(t('dialogs.bulkRenameFiles.success'))
+      regexpEl.value?.saveValueToHistory()
+      targetEl.value?.saveValueToHistory()
+    })
+    .catch(e => {
+      toast.error(e.toString())
+    })
+    .finally(() => {
+      running.value = false
+      contentStore.updateFileTree()
+      // close dialog, because haven't a good way to refresh torrent files after rename
+      // because bulkRenameFilesDialog can be create with subfolder
+      close()
+    })
+}
+
+const close = () => {
+  isOpened.value = false
+}
+
+watch([regexpInput, regexpFlagsInput, targetInput], () => {
+  dryRunRename()
+})
+
+onMounted(() => {
+  parseNode(props.node)
+})
+</script>
+
+<template>
+  <v-dialog v-model="isOpened" persistent>
+    <v-card density="compact">
+      <v-toolbar color="transparent">
+        <v-toolbar-title>{{ $t('dialogs.bulkRenameFiles.title') }}</v-toolbar-title>
+        <v-btn icon="mdi-close" @click="close()" />
+      </v-toolbar>
+      <v-card-text class="d-flex flex-column">
```

Comment:
Missing `v-card-title` element wiht ios safe margin
```suggestion
    <v-card density="compact">
      <v-card-title class="ios-margin">
        <v-toolbar color="transparent">
          <v-toolbar-title>{{ $t('dialogs.bulkRenameFiles.title') }}</v-toolbar-title>
          <v-btn icon="mdi-close" @click="close()" />
        </v-toolbar>
      </v-card-title>
      <v-card-text class="d-flex flex-column">
```